src/ia-bookreader/ia-bookreader.js

```diff
@@ -26,45 +32,101 @@ export class BookReader extends LitElement {
     this.bookreader = undefined;
     this.baseHost = 'https://archive.org';
     this.fullscreen = false;
+    this.signedIn = false;
     /** @type {ModalManager} */
     this.modal = undefined;
     /** @type {SharedResizeObserver} */
-    this.sharedObserver = new SharedResizeObserver();
+    this.sharedObserver = undefined;
+    this.loaded = false;
+    this.menuShortcuts = [];
+    this.menuContents = [];
   }
 
-  firstUpdated() {
-    this.createModal();
-  }
+  updated() {
+    if (!this.modal) {
+      this.modal = new ModalManager();
```

Comment:
Are you able to create a modal manager like this? I thought you had to add the DOM element.

Reply:
good catch, why did i think that this automagically unfolds itself... 🤔

Reply:
wow! just tried it and it looks good - as long as the `<modal-manager>` is in the DOM. so i will keep ^_^

Reply:
Do you mean the `new ModalManager()` only works if there's already a `<modal-manager>` element? Wouldn't this conditional only be entered if there's _not_ already a `<modal-manager>`?

If the case is that you already need a `<modal-manager>` element, you should be able to just query select it, ie: `this.modal = document.querySelector('modal-manager');`

Reply:
i am unsure if i want to reach up to the DOM like that..

>Wouldn't this conditional only be entered if there's not already a <modal-manager>?

hmm, ok, this is a stab at getting ready to receive a `ModalManager` thru dependency injection while working towards a bookreader that can just spin itself up (and the things that it needs) by itself.
(which i am clearly punting on until we have a plan around what the `<ia-bookreader>` looks like)

I guess i am not getting the difference between the two ways to instantiate the modal.

Reply:
thanks for checking in with me, currently resolving to - check for dom, if not, load.